contents/tutorials/hogql-breakdowns.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Using HogQL for advanced breakdowns
+date: 2023-05-16
+author: ["ian-vanagas"]
+showTitle: true
+sidebar: Docs
+featuredImage: ../images/tutorials/banners/tutorial-8.png
+tags: ['hogql', 'trends']
+---    
+
+HogQL opens limitless possibilities for how you can breakdown your trends, funnels, and more. This tutorial showcases some of the advanced breakdowns you can create using HogQL.
+
+To create a breakdown using HogQL, create an insight then under "Breakdown by," click "Add breakdown," select HogQL from the options, and add your expression.
+
+> To understand the full possibilities of HogQL, check out the [available functions in our docs](/docs/product-analytics/hogql#supported-clickhouse-functions) as well as the [events](https://app.posthog.com/data-management/events) and [properties](https://app.posthog.com/data-management/properties) lists from your PostHog instance.
+
+## All events
+
+HogQL includes the "All events" series, which can break down in many ways, including by individual events. This enables you to visualize the breakdown of all your events at once. 
+
+To do this, search for the "All events" trend, then breakdown with the HogQL expression `event`.
+
+![Events](../images/tutorials/hogql-breakdowns/events.png)
+
+You could also breakdown all events by `person_id` to find your most active users.
+
+## Replacing repetitive values
+
+Many strings, such as URLs, include repetitive values combined with unique values. It is common to want to replace that unique value with a placeholder value to combine the repetitive values into one for analysis.
+
+For example, you have a path that uses a specific user’s UUID like `/users/123e4567-e89b-12d3-a456-426655440000`. You can use `replaceRegexpOne` to replace `123e4567-e89b-12d3-a456-426655440000` with `:uuid` . The HogQL expression for this looks like this:
+
+```
+replaceRegexpOne(
+	properties.$pathname,
+	'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}',
+	':uuid'
+)
+```
+
+Another way to replace a repetitive value is to use `replaceOne`. If we wanted to replace every instance of `https://app.posthog.com` with `/`, we can do:
+
+```
+replaceOne(properties.$current_url, 'https://app.posthog.com', '/')
+```
+
+> **Note:** Make sure you use straight apostrophes (`'`) for string literals and backticks (\`) or quotes (`"`) for database identifiers in your expressions. **Don't use** curly apostrophes (`’`) or curly quotes (`“` or `”`).
```

Comment:
```suggestion
> **Note:** Make sure you use straight apostrophes (`'`) for string literals and backticks (`\`) or quotes (`"`) for database identifiers in your expressions. **Don't use** curly apostrophes (`’`) or curly quotes (`“` or `”`).
```

Why?

Reply:
They just aren't part of the query language basically. It doesn't recognize them.